SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/DispatcherTest.cpp

```diff
@@ -0,0 +1,148 @@
+{
+	SpatialGDK::ComponentData Data = SpatialGDK::CreateTestComponentData(ComponentId, Value);
+	SpatialGDK::EntityViewElement Element;
+	Element.Components.Emplace(MoveTemp(Data));
+	return Element;
+}
+} // anonymous namespace
+
+//
+// Tests begin here
+//
+
+DISPATCHER_TEST(GIVEN_Dispatcher_WHEN_Callback_Added_Then_Invoked_THEN_Callback_Invoked_With_Correct_Values)
+{
+	bool Invoked = false;
+	SpatialGDK::FDispatcher Dispatcher;
+
+	const SpatialGDK::FComponentValueCallback Callback = [&Invoked](const SpatialGDK::FEntityComponentChange Change) {
+		if (Change.EntityId == ENTITY_ID && Change.Change.ComponentId == COMPONENT_ID
+			&& GetValueFromSchemaComponentData(Change.Change.Data) == COMPONENT_VALUE)
+		{
+			Invoked = true;
+		}
+	};
+
+	Dispatcher.RegisterComponentAddedCallback(COMPONENT_ID, Callback);
+	SpatialGDK::ComponentChange Change = CreateComponentAddedChange(COMPONENT_ID, COMPONENT_VALUE);
+	Dispatcher.InvokeCallbacks(CreateComponentAddedDeltaWithChange(ENTITY_ID, Change));
+
+	TestTrue("Callback was invoked", Invoked);
+
+	// Now a few more times, but with incorrect values, just in case
+	Invoked = false;
+
+	Change = CreateComponentAddedChange(COMPONENT_ID, OTHER_COMPONENT_VALUE);
+	Dispatcher.InvokeCallbacks(CreateComponentAddedDeltaWithChange(ENTITY_ID, Change));
+	TestFalse("Callback was not invoked", Invoked);
+
+	Change = CreateComponentAddedChange(OTHER_COMPONENT_ID, COMPONENT_VALUE);
+	Dispatcher.InvokeCallbacks(CreateComponentAddedDeltaWithChange(ENTITY_ID, Change));
+	TestFalse("Callback was not invoked", Invoked);
+
+	Change = CreateComponentAddedChange(COMPONENT_ID, COMPONENT_VALUE);
+	Dispatcher.InvokeCallbacks(CreateComponentAddedDeltaWithChange(OTHER_ENTITY_ID, Change));
+	TestFalse("Callback was not invoked", Invoked);
+
+	return true;
+}
+
+DISPATCHER_TEST(GIVEN_Dispatcher_With_Callback_WHEN_Callback_Removed_THEN_Callback_Not_Invoked)
+{
+	bool Invoked = false;
+	SpatialGDK::FDispatcher Dispatcher;
+
+	const SpatialGDK::FComponentValueCallback Callback = [&Invoked](SpatialGDK::FEntityComponentChange) {
+		Invoked = true;
+	};
+
+	const SpatialGDK::CallbackId Id = Dispatcher.RegisterComponentAddedCallback(COMPONENT_ID, Callback);
+	const SpatialGDK::ComponentChange Change = CreateComponentAddedChange(COMPONENT_ID, COMPONENT_VALUE);
+	const TArray<SpatialGDK::EntityDelta> Deltas = CreateComponentAddedDeltaWithChange(ENTITY_ID, Change);
+	Dispatcher.InvokeCallbacks(Deltas);
+
+	TestTrue("Callback was invoked", Invoked);
+
+	Invoked = false;
+	Dispatcher.RemoveCallback(Id);
+	Dispatcher.InvokeCallbacks(Deltas);
+
+	TestFalse("Callback was not invoked again", Invoked);
+
+	return true;
+}
+
+DISPATCHER_TEST(GIVEN_Dispatcher_WHEN_Callback_Added_And_Invoked_THEN_Callback_Invoked_With_Correct_Values)
+{
+	bool Invoked = false;
+	SpatialGDK::FDispatcher Dispatcher;
+	const TUniquePtr<SpatialGDK::EntityView> View = MakeUnique<SpatialGDK::EntityView>();
```

Comment:
This doesn't look like this needs to be a raw pointer. Just making and EntityView should do fine.

Reply:
done!